deployments/testpoa/docker-compose.yml

```diff
@@ -0,0 +1,147 @@
+version: '3.5'
+services:
+  poa-node-arthur: &poa-node
+    build:
+      context: .
+      dockerfile: ./OpenEthereum.Dockerfile
+      args:
+        ETHEREUM_HASH: 9838f59b6536e7482e145fa55acf07ac4e824ed0
+        BRIDGE_HASH: master
+    entrypoint:
+      - /home/openethereum/openethereum
+      - --config=/config/poa-node-config
+      - --node-key=arthur
+      - --engine-signer=0x005e714f896a8b7cede9d38688c1a81de72a58e4
+    volumes:
+      - ./poa-config:/config
+    environment:
+      RUST_LOG: rpc=trace,txqueue=trace
+    ports:
+      - "8545:8545"
+      - "8546:8546"
+      - "30303:30303"
+
+  poa-node-bertha:
+    <<: *poa-node
+    entrypoint:
+      - /home/openethereum/openethereum
+      - --config=/config/poa-node-config
+      - --node-key=bertha
+      - --engine-signer=0x007594304039c2937a12220338aab821d819f5a4
+    ports:
+      - "8645:8545"
+      - "8646:8546"
+      - "31303:30303"
+
+  poa-node-carlos:
+    <<: *poa-node
+    entrypoint:
+      - /home/openethereum/openethereum
+      - --config=/config/poa-node-config
+      - --node-key=carlos
+      - --engine-signer=0x004e7a39907f090e19b0b80a277e77b72b22e269
+    ports:
+      - "8745:8545"
+      - "8746:8546"
+      - "32303:30303"
+
+  relay-eth2sub:
+    build:
+      context: .
+      dockerfile: ../../Dockerfile
+      args:
+        BRIDGE_HASH: master
+        PROJECT: ethereum-poa-relay
+    entrypoint: /config/relay-eth2sub-entrypoint.sh
+    volumes:
+      - ./bridge-config:/config
+    environment:
+      RUST_LOG: rpc=trace,bridge=trace
+    depends_on: &all-nodes
+      - poa-node-arthur
+      - poa-node-bertha
+      - poa-node-carlos
+      - bridge-node-alice
+      - bridge-node-bob
+      - bridge-node-charlie
+
+  # relay-sub2eth:
+  #   build:
+  #     context: .
+  #     dockerfile: ../../Dockerfile
+  #     args:
+  #       BRIDGE_HASH: master
+  #       PROJECT: ethereum-poa-relay
+  #   entrypoint: /config/relay-sub2eth-entrypoint.sh
+  #   volumes:
+  #     - ./bridge-config:/config
+  #   environment:
+  #     RUST_LOG: rpc=trace,bridge=trace
+  #   depends_on: *all-nodes
+
+  bridge-node-alice: &bridge-node
+    build:
+      context: .
+      dockerfile: ../../Dockerfile
```

Comment:
This path didn't work out of the box for me. I kept getting the following error

```
Building bridge-node-alice
ERROR: Forbidden path outside the build context: ../../Dockerfile ()
```

It was only after changing it to

```yaml
context: ../../
dockerfile: ./Dockerfile
```

that I managed to get these built.

Reply:
I rather duplicated the file, having the entire repo in a context for that specific deployment is a big wasteful. The top-level `Dockerfile` is now using local files instead of checking out the repo from GH.

Reply:
I think that's reasonable!